source/analysis/OLDSGM/SGM1DFastScanFilterAB.cpp

```diff
@@ -0,0 +1,241 @@
+	if(dataSources.count() == 1 && dataSources.at(0)->rank() == 1)
+		return true;
+
+	return false;
+}
+
+// Set the data source inputs.
+void SGM1DFastScanFilterAB::setInputDataSourcesImplementation(const QList<AMDataSource*>& dataSources) {
+
+	// disconnect connections from old source, if it exists.
+	if(inputSource_) {
+		disconnect(inputSource_->signalSource(), SIGNAL(valuesChanged(AMnDIndex,AMnDIndex)), this, SLOT(onInputSourceValuesChanged(AMnDIndex,AMnDIndex)));
+		disconnect(inputSource_->signalSource(), SIGNAL(sizeChanged(int)), this, SLOT(onInputSourceSizeChanged()));
+		disconnect(inputSource_->signalSource(), SIGNAL(stateChanged(int)), this, SLOT(onInputSourceStateChanged()));
+	}
+
+	if(dataSources.isEmpty()) {
+		inputSource_ = 0;
+		sources_.clear();
+
+		inputAxis_ = AMAxisInfo("invalid", 0, "No input data");
+		axes_[0] = AMAxisInfo("invalid", 0, "No input data");
+		setDescription("-- No input data --");
+		invalidateCache();
+	}
+
+	// we know that this will only be called with valid input source
+	else {
+		inputSource_ = dataSources.at(0);
+		sources_ = dataSources;
+		inputAxis_ = inputSource_->axisInfoAt(0);
+		axes_[0] = inputSource_->axisInfoAt(0);
+
+
+		setDescription(QString("FastScan Filter of %1")
+					   .arg(inputSource_->name()));
+
+		connect(inputSource_->signalSource(), SIGNAL(valuesChanged(AMnDIndex,AMnDIndex)), this, SLOT(onInputSourceValuesChanged(AMnDIndex,AMnDIndex)));
+		connect(inputSource_->signalSource(), SIGNAL(sizeChanged(int)), this, SLOT(onInputSourceSizeChanged()));
+		connect(inputSource_->signalSource(), SIGNAL(stateChanged(int)), this, SLOT(onInputSourceStateChanged()));
+
+	}
+
+	reviewState();
+
+	emitSizeChanged(0);
+	emitValuesChanged();
+	emitAxisInfoChanged(0);
+	emitInfoChanged();
+}
+
+bool SGM1DFastScanFilterAB::values(const AMnDIndex &indexStart, const AMnDIndex &indexEnd, double *outputValues) const
+{
+	if(indexStart.rank() != 1 || indexEnd.rank() != 1)
+		return false;
+
+	if(!isValid())
+		return false;
+
+#ifdef AM_ENABLE_BOUNDS_CHECKING
+	if((unsigned)indexEnd.i() >= (unsigned)axes_.at(0).size || (unsigned)indexStart.i() > (unsigned)indexEnd.i())
+		return false;
+#endif
+
+	int totalSize = indexStart.totalPointsTo(indexEnd);
+
+	if (!cacheCompletelyInvalid_){
+
+		int offset = indexStart.i();
+
+		for (int i = 0; i < totalSize; i++)
+			outputValues[i] = double(cachedValues_.at(i+offset));
+	}
+
+	else{
+
+		for (int i = 0; i < totalSize; i++)
+			outputValues[i] = 27.27;
+	}
+
+	return true;
+}
+
+AMNumber SGM1DFastScanFilterAB::value(const AMnDIndex& indexes) const{
+	if(indexes.rank() != 1)
+		return AMNumber(AMNumber::DimensionError);
+
+	if(!isValid())
+		return AMNumber(AMNumber::InvalidError);
+
+#ifdef AM_ENABLE_BOUNDS_CHECKING
+	if((unsigned)indexes.i() >= (unsigned)axes_.at(0).size)
+		return AMNumber(AMNumber::OutOfBoundsError);
+#endif
+
+	int index = indexes.i();
+
+	if(!cacheCompletelyInvalid_)
+		return cachedValues_.at(index);
+
+	return AMNumber(27.27);
```

Comment:
This is in OldSGM, so it isn't used any more. All the classes which live in this subdir will be deleted soonish.